src/mon/PGMap.cc

```diff
@@ -3198,6 +3210,48 @@ void PGMap::get_health_checks(
   }
 }
 
+const uint64_t PGMap::get_num_omap_bytes_per_osd(int osd, uint64_t* const total) const
+{
+  uint64_t ret = 0;
+  auto p = pg_by_osd.find(osd);
+
+  if (p == pg_by_osd.end()) {
+    return 0;
+  }
+  for (auto pgid : p->second) {
+    auto stat = pg_stat.at(pgid);
+    auto it = pool_size.find(pgid.pool());
+    unsigned size = 1;
+    if (it != pool_size.end()) {
+      size = it->second;
+    }
+    ret += stat.stats.sum.num_omap_bytes / size; // Divide by number of replicas
```

Comment:
>  sizes into the pgmap, which doesn't seem to be complete in this commit?)

How is it incomplete Greg? not sure I follow?

I'll discuss the rest with Josh tomorrow since we already went over this once.

Reply:
Maybe there’s another PR associated with this (since it just says “test” in the title), but despite adding pool size to the PGMao and Incremental, nothing fills them in yet.

Reply:
@gregsfortytwo Doesn't line 1228 do that Greg?

Reply:
Yes, it copies the Incremental into the PGMap. But...oh, I have missed the PGMapUpdater function below, never mind!

Reply:
I was beginning to wonder how it worked!